shpc/main/modules/__init__.py

```diff
@@ -409,43 +396,38 @@ def write_version_file(self, uri, tag, default_choice=False):
 
         # First or last installed
         elif self.settings.default_version == "last_installed":
-            self._default_version_last_installed(version_file, tag)
+            self._set_default_version(version_file, tag)
 
         elif self.settings.default_version == "first_installed":
-            self._default_version_first_installed(version_file, tag, default_choice)
+            first_installed = True if len(os.listdir(version_dir)) == 1 else False
+            if first_installed:
+                self._set_default_version(version_file, tag)
 
     def update_version_file(self, version_dir):
         """
         Given a module directory, update the version file.
         """
-        # The module directory has the version (which is deleted)
-        version_file = os.path.join(version_dir, ".version")
-
-        # If a version file doesn't exist, nothing to update
-        if not os.path.exists(version_file):
-            return
-
+        # Count how many versions we actually have
         found = [x for x in os.listdir(version_dir) if x != ".version"]
 
         # No versions left, remove the whole directory
         if len(found) == 0:
             self._cleanup(version_dir)
             return
 
-        uri = os.path.basename(version_dir)
+        # With the following settings, .version doesn't contain an actual version
+        # so there is nothing to update
```

Comment:
This isn’t right - if we are updating an install with default version true we need to change it to the last installed version.

Reply:
`default_version=true` means we let Lmod/TCL decide what the default version should be. For Lmod that's done by having a `.version` file that is empty or refers to a version that doesn't exist - so no need to update. For TC:, that's done by _not_ having a `.version` file at all

Reply:
Right - but for the design of this main class it should be calling the appropriate function and not making assumptions or decisions early about the module software (that was the idea of the design having these sub-functions that can optionally be implemented by each of tcl and lmod.

Reply:
Sure. I can perhaps make this function more similar to `write_version_file` by adding the series of `if self.settings.default_version ==`. I'll give it a go and will let you know